### PR TITLE
RichText: absorb internal list controls

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This change aims to absorb any RichText related logic from the list block into the RichText component. With this, none of our core blocks will use the `unstableGetSettings` or `unstableOnSetup` props, which I would like to remove entirely so that no TinyMCE APIs remain exposed.

Also fixes a bug where the the active button is not set for nested lists.

After #10799, the list buttons may be rewritten to manipulate the internal rich text value directly, removing the dependency on the TinyMCE lists plugin.

## How has this been tested?
The list block should continue to work as before.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by aduth** [comment]
>With this, none of our core blocks will use the `unstableGetSettings` or `unstableOnSetup` props, which I would like to remove entirely so that no TinyMCE APIs remain exposed.

Can we delete them here and now?

**Comment by iseulde** [comment]
> Can we delete them here and now?

Sure thing! More than happy to!

**Comment by iseulde** [comment]
There you go!

File: packages/editor/src/components/rich-text/tinymce.js [hunk 1]
```diff
@@ -191,7 +192,16 @@ export default class TinyMCE extends Component {
...
-		} );
+			forced_root_block: multilineTag || false,
+			// Allow TinyMCE to keep one undo level for comparing changes.
+			// Prevent it otherwise from accumulating any history.
+			custom_undo_redo_levels: 1,
+		};
+
+		if ( multilineTag === 'li' ) {
+			settings.plugins.push( 'lists' );
+			settings.lists_indent_on_tab = true;
```
Comment (tofumatt) [inline packages/editor/src/components/rich-text/tinymce.js]: ❤️

File: packages/editor/src/components/rich-text/list-edit.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+export const ListEdit = ( { editor, onTagNameChange, tagName } ) => (
+	<Fragment>
+		<RichTextShortcut
+			type="primary"
+			character="]"
+			onUse={ () => editor.execCommand( 'Outdent' ) }
+		/>
+		<RichTextShortcut
+			type="primary"
+			character="["
```
Comment (aduth) [inline packages/editor/src/components/rich-text/list-edit.js]: These were ported backwards. `]` should be for the indent.

**Review by aduth — DISMISSED** [review]

File: packages/editor/src/components/rich-text/list-edit.js (hunk 2 above)
Comment (aduth) [reply packages/editor/src/components/rich-text/list-edit.js]: > These were ported backwards. `]` should be for the indent.

Updated in 25997a2

**Comment by aduth** [comment]
There's something not working quite right with indentation on new list items. I'll plan to investigate.

**Comment by aduth** [comment]
It's not immediately obvious to me what's going on with the list indentation. The callbacks for the key handlers are not being called. Doing anything after the new item (even moving mouse, leaving screen and returning, pressing Escape) is enough to cause the keybind to work again, which itself is concerning as these don't seem like they should be causing any change in state to the RichText element.

We also still didn't quite port the existing binds as they were before, because Cmd+M / Cmd+Shift+M were only bound previously if the keyboard didn't support the `]`.

Given this is more a refactoring, and while I'd like to see the unstable functions gone sooner than later, I'm inclined to punt this to 4.4.

**Comment by iseulde** [comment]
@aduth Could you provide some steps to reproduce the indentation issue? I don't find any issues.

**Comment by iseulde** [comment]
Tbh the shortcuts added for lists are not great. On a Mac, 3/4 of the shortcuts clash:

screen shot 2018-11-14 at 12 37 43
screen shot 2018-11-14 at 12 38 39

**Comment by iseulde** [comment]
@aduth Okay, I found the problem with the shortcuts. They unmount when the toolbar disappears. So when when you type, they no longer work, when you hover again so the toolbars show, they work again.

**Comment by iseulde** [comment]
> We also still didn't quite port the existing binds as they were before, because Cmd+M / Cmd+Shift+M were only bound previously if the keyboard didn't support the ].

It's a bit strange to me that we're only binding based on some random language detection. The [] shortcuts won't do anything if you don't have the keys on your keyboard so it doesn't hurt to bind them. I don't think we can accurately detect if a keyboard has these keys

I also think we should bind Cmd+M / Cmd+Shift+M for everyone. Why would we only make them work in certain cases?

**Review by aduth — APPROVED** [review]
Looks good 👍

**Comment by iseulde** [comment]
Thanks!

**Pull request merged**